components/lib/datatable/datatable.d.ts

```diff
@@ -358,20 +383,20 @@ export interface DataTableProps extends Omit<React.DetailedHTMLProps<React.Input
     onRowToggle?(e: DataTableRowToggleParams): void;
     onRowUnselect?(e: DataTableUnselectParams): void;
     onSelectAllChange?(e: DataTableSelectAllChangeParams): void;
-    onSelectionChange?(e: DataTableSelectionChangeParams): void;
+    onSelectionChange?(e: DataTableSelectionChangeParams<TValue>): void;
     onSort?(e: DataTablePFSEvent): void;
     onStateRestore?(state: object): void;
     onStateSave?(state: object): void;
     onValueChange?(value: any[]): void;
-    rowClassName?(data: any, options: DataTableRowClassNameOptions): object | string;
-    rowEditValidator?(data: any, options: DataTableRowEditValidatorOptions): boolean;
+    rowClassName?(data: any, options: DataTableRowClassNameOptions<TValue>): object | string;
+    rowEditValidator?(data: any, options: DataTableRowEditValidatorOptions<TValue>): boolean;
     rowExpansionTemplate?(data: any, options: DataTableRowExpansionTemplate): React.ReactNode;
-    showRowReorderElement?(data: any, options: DataTableShowRowReorderElementOptions): boolean | undefined | null;
-    showSelectionElement?(data: any, options: DataTableShowSelectionElementOptions): boolean | undefined | null;
+    showRowReorderElement?(data: any, options: DataTableShowRowReorderElementOptions<TValue>): boolean | undefined | null;
+    showSelectionElement?(data: any, options: DataTableShowSelectionElementOptions<TValue>): boolean | undefined | null;
```

Comment:
For example in the above I believe `data: any` should be `data: DataTableValue` no?

Reply:
I have not used that function before but I could change it to `DataTableValue`  if you think its correct. There are multiple functions with `data: any` which I am not sure what to type. There are also cases where some types are `any[] | ...` and I am not sure what purpose those serve

Reply:
In most cases if you look a the source code any Datatable function passing `data: any` is USUALLY 1 row of data from the table which is your `DataTableValue` singular definition.